### PR TITLE
Custom Confirm Modal. Refactorization

### PR DESCRIPTION
**Pull request by KamBustos**
I am creating a separate PR because the last PR is severely outdated and I would like to add my own implementations for this. For Sprint 3 I am picking up this issue that Josh was working on. So far, I have merged changes from the main, and I plan to refine and add to Josh's preexisting work for the remainder of Sprint 3. In short, this issue entails adding a custom confirm modal, which should be implemented in all previous places where native browser confirm was used. This issue has not been worked on for a while, so there are many merge conflicts to resolve. I am working on the merge conflicts with Abhilash. This is what the latest change looks like, but when clicking "OK", the witness is not deleted:

Dialogue:
image

Confirmation:
image

### CODE REVIEW

**Review by AbhilashKotha — COMMENTED** [review]
Works well for me on delete actions ( for both true and false conditions) The delete witness does not work, but it does not work on the main branch too!

**Review by thehabes — CHANGES_REQUESTED** [review]
I am a little worried that I see so many code changes as part of the diff that were a part of the delete refactor.  You may need to merge with `main` to get a clean diff, but I guess it's OK to move forward with it.  You will really have to keep an eye on this merge @AbhilashKotha 

This does appear to function as expected and the confirm() functionality of returning true or false is respected.

A couple of changes to make.  Since you are essentially making a custom HTML, scope your querySelectors to your HTML instead of the whole `document`.  You should be good to go after that.

I will also note that I was able to confirm() and delete a Witness.  I'll still double check on that delete after we merge.

**Review by thehabes — APPROVED** [review]

**Review by AbhilashKotha — APPROVED** [review]
lgtm

**Pull request merged**
